src/Chainweb/BlockHeader/Genesis.hs

```diff
@@ -101,12 +103,20 @@ genesisBlockPayload :: ChainwebVersion -> ChainId -> PayloadWithOutputs
 genesisBlockPayload Test{} _ = emptyPayload
 genesisBlockPayload TimedConsensus{} _ = emptyPayload
 genesisBlockPayload PowConsensus{} _ = emptyPayload
-genesisBlockPayload TimedCPM{} _ = TN.payloadBlock
-genesisBlockPayload FastTimedCPM{} _ = TN.payloadBlock
+genesisBlockPayload TimedCPM{} _ = TNN.payloadBlock
+genesisBlockPayload FastTimedCPM{} cid = case chainIdInt @Int cid of
+    0 -> TN0.payloadBlock
+    _ -> TNN.payloadBlock
```

Comment:
This needs allocations too?

Reply:
Yes - we're using this for our e2e tests to avoid mining